auto-publish/utils.js

```diff
@@ -0,0 +1,231 @@
+const child_process = require("child_process");
+const semver = require("semver");
+const path = require("path");
+const fs = require("fs-extra");
+const OUR_RC_DIST_TAG = "rc";
+const basePackageJSON = require("../source/package.base.json");
+const {
+  default: { patch }
+  // @ts-ignore
+} = require("react-hot-loader/webpack");
```

Comment:
`react-hot-loader` then also have to be auto-updated.

Reply:
We may track the version change of react-hot-loader & react-dom, and make new version when each of them changes,
But which version number should we give our artifact? we won't be able to strictly follow react-dom no more

Reply:
16.8.6-xxxx?

Reply:
you mean:
`react.dom.version-rhl.version` ? and then we can split  on `-` and see the version in rhl or dom have changed?

It means we somewhat lose semver semantics, but i can't think about any other good option to save state of last released versions

Reply:
Done, and please see how the new code handling the version